### DISCUSSION TITLE
Need next route for unstable_useBlocker and unstable_usePrompt

### DISCUSSION THREAD

**Discussion by resatDev** [remix-run/react-router#10858] [Proposals]
After the `Prompt component` was removed, the `unstable_useBlocker` and `unstable_usePrompt` hooks were **added again**. However, the **"message"** prop, which is one of the most important features of the Prompt component, **does not have the ability to work as a function**. One of the parameters that the function **message prop return was the next route to proceed**. I will need this as soon as possible. The **nextPathname prop used in the unstable_useBlocker** hooks is the related prop which I need.

**Comment by brophdawg11** [comment]
The primary way to block navigation in `RouterProvider` is `unstable_useBlocker`, which gives you control over the UI and avoids messy edge cases that come with `usePrompt` (since browsers handle back/forward navigations while the confirm dialog is open quite differently).  We do not have plans to enhance `unstable_usePrompt` at this time - it's only about 20 lines of code wrapping `unstable_useBlocker` so your best path forward at this time would be to copy it and modify it to suit your needs - specifically by passing a function to `useBlocker` instead of a boolean.
